sound/soc/sof/pm.c

```diff
@@ -204,8 +204,11 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 
 	target_state = snd_sof_dsp_power_target(sdev);
 
-	/* Skip to platform-specific suspend if DSP is entering D0 */
-	if (target_state == SOF_DSP_PM_D0)
+	/*
+	 * Skip to platform-specific suspend if DSP is entering D0 or if it has
+	 * crashed
+	 */
+	if (target_state == SOF_DSP_PM_D0 || sdev->fw_state == SOF_FW_CRASHED)
```

Comment:
not sure about this one. In the 'sof_tear_down_pipelines' below we free some widgets. We wouldn't be able to restart with a clean slate out of D3 if we skip this part.

@ranj063 you'll have to chime in on this change,